perses/tests/test_geometry_engine.py

```diff
@@ -26,14 +26,553 @@
+        return internals
+
+    @internal_coordinates.setter
+    def internal_coordinates(self, internal_coordinates):
+        internals_without_units = np.zeros(3, dtype=np.float64)
+        internals_without_units[0] = internal_coordinates[0].value_in_unit(unit.nanometer)
+        internals_without_units[1] = internal_coordinates[1].value_in_unit(unit.radians)
+        internals_without_units[2] = internal_coordinates[2].value_in_unit(unit.radians)
+        positions_without_units = self._positions.value_in_unit(unit.nanometer)
+        new_cartesian_coordinates = coordinate_numba.internal_to_cartesian(positions_without_units[1], positions_without_units[2], positions_without_units[3], internals_without_units)
+        self._positions[0] = unit.Quantity(new_cartesian_coordinates, unit=unit.nanometer)
+
+    @property
+    def bond_parameters(self):
+        return (self._default_r0, self._default_bond_k)
+
+    @property
+    def angle_parameters(self):
+        return (self._default_angle_theta0, self._default_angle_k)
+
+    @property
+    def torsion_parameters(self):
+        return (self._default_torsion_periodicity, self._default_torsion_phase, self._default_torsion_k)
+
+def test_propose_angle():
+    """
+    Test the proposal of angles by GeometryEngine by comparing to proposals from a normal distribution
+    with mean theta0 (equilibrium angle) and variance sigma = sqrt(1.0/(k*beta)), where k is the force
+    constant and beta is the inverse temperature. A Kolmogorov-Smirnov test is used for that comparison.
+    """
+    from perses.rjmc.geometry import FFAllAngleGeometryEngine
+    import scipy.stats as stats
+    geometry_engine = FFAllAngleGeometryEngine()
+
+    #Create a test system with only an angle force
+    testsystem = FourAtomValenceTestSystem(bond=False, angle=True, torsion=False)
+
+    #There is only one angle in this system--extract it
+    angle = testsystem.structure.angles[0]
+    angle_with_units = geometry_engine._add_angle_units(angle)
+
+    #extract the parameters and convert them to the equivalents for a normal distribution
+    #without units
+    (theta0, k) = testsystem.angle_parameters
+    sigma = unit.sqrt(1.0/(beta*k))
+    sigma_without_units = sigma.value_in_unit(unit.radian)
+    theta0_without_units = theta0.value_in_unit(unit.radian)
+
+    #allocate an array for proposing angles from the appropriate distribution
+    angle_array = np.zeros(1000)
+    for i in range(1000):
+        proposed_angle_with_units = geometry_engine._propose_angle(angle_with_units, beta)
+        angle_array[i] = proposed_angle_with_units.value_in_unit(unit.radians)
+
+    #compare the sampled angles to a normal cdf with the appropriate parameters using
+    #the Kolomogorov-Smirnov test. The null hypothesis is that they are drawn from the same
+    #distribution (the test passes).
+    (dval, pval) = stats.kstest(angle_array,'norm', args=(theta0_without_units, sigma_without_units))
+    if pval < 0.05:
```

Comment:
I don't think we should not be using p-values of 0.05 in our significance testing. Even if everything is working correctly, each test will randomly fail 5% of the time, and as the number of these tests increases, the probability of encountering a test failure approaches unity, making the tests useless. 

I'd love to discuss the optimal testing strategy for these stochastic tests, but my strong preferences is to select a p-value that is highly significant. I generally prefer something like:
```
pval_base / (ntests * ncommits)
```
where `pval_base` is a highly significant p-value (not 0.05, much smaller), `ntests` is the number of statistical tests, and `ncommits` is the number of commits we expect in the lifetime of the project (say, 5000). This essentially applies a multiple hypothesis testing correction such that when a test failure occurs, we are highly confident something is actually wrong and we need to investigate what is wrong.